src/pipecat/services/asyncai/tts.py

```diff
@@ -0,0 +1,513 @@
+                "encoding": encoding,
+                "sample_rate": sample_rate,
+            },
+            "language": self.language_to_service_language(params.language)
+            if params.language
+            else "en",
+        },
+        
+        self.set_model_name(model)
+        self.set_voice(voice_id)
+        self._global_context_id = str(uuid.uuid4())
+
+        self._context_id = None
+        self._receive_task = None
+        self._keepalive_task = None
+
+    def can_generate_metrics(self) -> bool:
+        """Check if this service can generate processing metrics.
+
+        Returns:
+            True, as Async service supports metrics generation.
+        """
+        return True
+
+    def language_to_service_language(self, language: Language) -> Optional[str]:
+        """Convert a Language enum to Async language format.
+
+        Args:
+            language: The language to convert.
+
+        Returns:
+            The Async-specific language code, or None if not supported.
+        """
+        return language_to_async_language(language)
+
+    def _build_msg(
+        self, text: str = "", force: bool = False
+    ) -> str:
+        msg = {
+            "transcript": text,
+            "force": force
+        }
+        return json.dumps(msg)
+    
+    async def start(self, frame: StartFrame):
+        """Start the Async TTS service.
+
+        Args:
+            frame: The start frame containing initialization parameters.
+        """
+        await super().start(frame)
+        await self._connect()
+
+    async def stop(self, frame: EndFrame):
+        """Stop the Async TTS service.
+
+        Args:
+            frame: The end frame.
+        """
+        await super().stop(frame)
+        await self._disconnect()
+
+    async def cancel(self, frame: CancelFrame):
+        """Cancel the Async TTS service.
+
+        Args:
+            frame: The cancel frame.
+        """
+        await super().cancel(frame)
+        await self._disconnect()
+
+    async def _connect(self):
+        await self._connect_websocket()    
+
+        if self._websocket and not self._receive_task:
+            self._receive_task = self.create_task(self._receive_task_handler(self._report_error))
+
+        if self._websocket and not self._keepalive_task:
+            self._keepalive_task = self.create_task(self._keepalive_task_handler())
+
+    async def _disconnect(self):
+        if self._receive_task:
+            await self.cancel_task(self._receive_task)
+            self._receive_task = None
+        
+        if self._keepalive_task:
+            await self.cancel_task(self._keepalive_task)
+            self._keepalive_task = None
+
+        await self._disconnect_websocket()
+
+    async def _connect_websocket(self):
+        try:
+            if self._websocket and self._websocket.open:
+                return
+            logger.debug("Connecting to Async")
+            self._websocket = await websockets.connect(
```

Comment:
Update for asyncio implementation:
```suggestion
            self._websocket = await websocket_connect(
```